CONTRIBUTING.md

```diff
@@ -136,6 +136,55 @@ This will start a development server of the backend that will allow you to brows
 
 Because we are using FastAPI the Karman backend has an auto-generated API documentation for almost all endpoints. You can open the documentation by running the development server and navigating to http://localhost:8000/v1/docs or http://localhost:8000/v1/redoc. Note the `v1` in the path.
 
+## Database Setup
+
+In order to be able to run the backend you need a database connection. By default Karman is configured to use an SQLite database in the `db.sqlite` file. This is usually fine for development but can be changed via the `db_url` setting.
+
+To work with databases you need the appropriate database drivers. These can be installed as follows:
+
+```shell
+# SQLite (easiest for development)
+poetry install --extras "sqlite"
+# MySQL / MariaDB
+poetry install --extras "mysql"
+# PostgreSQL
+poetry install --extras "postgresql"
+```
+
+### Creating migrations
+
+Whenever you modify a database model we need to migrate the database schema. For this purpose we use [Alembic](https://alembic.sqlalchemy.org). Giving a full introduction into Alembic is outside the scope of this document so we can only cover the most basic cases.
+
+After you make changes to a model, you can automatically create migrations by running the following command:
+
+```shell
+poe makemigration "<message>"
+```
+
+`<message>` is intended for a concise description of the changes made and will be used as part of the migration filename.
+
+In basic cases the auto generated migrations might fine but in many cases it is necessary to edit the migration files manually afterwards.
```

Comment:
spelling mistake: "might be fine"

To which extend do the migration files have to be changed manually afterwards?

Reply:
Updated. For the usual cases (adding models, adding model fields, adding constraints or indexes) the automatic migrations should be fine. Apart from being unable to detect changes in table or column names the limitations should be pretty similar to the ones in Django I think. For example I think that if we change the type of a column depending on the type it might be necessary to take care that the migrations are able to migrate the data. This very much depends on the actual change.

Practically speaking I don't expect too much hassle with this in the near future. This only becomes relevant if we decide to heavily change the data model down the road (in which case a manually written migration might make sense anyways).

One case that is not very well covered by Alembic is ingesting rows into tables during migrations. This is not terribly useful for most models but if we were to implement Permissions in a similar way to Django this might be necessary. I have however written an Alembic extension for that case if it becomes necessary ([Link](https://github.com/Karaoke-Manager/backend/tree/9821927af826a850f0287051060d0d96cab947ef/extensions/alembic-data)).